File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        l =0 
        uniqueSet = set()
        res = 0

        for r in range(len(s)):
            while s[r] in uniqueSet:
                uniqueSet.remove(s[l])
                l+=1
            uniqueSet.add(s[r])
            res = max(res, r-l+1)

        return res

File: Data Structures & Algorithms/max-water-container/submission-0.java
class Solution {
    public int maxArea(int[] heights) {
        int l = 0;
        int r = heights.length-1;
        int area = 0;
        int res = 0;

        while (l<r){

            int width = r-l;
            int height = Math.min(heights[l], heights[r]);
            
            area = width*height;
            res = Math.max(area, res);

            if(heights[l]<heights[r]){
                l++;
            }
            else{
                r--;
            }

            
        }

        return res;

    }
}
